src/Chapter3.hs

```diff
@@ -970,7 +1189,30 @@ implement the following functions:
 
 🕯 HINT: to implement this task, derive some standard typeclasses
 -}
-
+data DayOfWeek
+    = Monday 
+    | Tuesday 
+    | Wednesday 
+    | Thursday 
+    | Friday
+    | Saturday
+    | Sunday
+    deriving(Show, Enum, Bounded)
+
+isWeekend :: DayOfWeek -> Bool
+isWeekend (Saturday) = True
+isWeekend (Sunday) = True
+isWeekend (_) = False
+
+nextDay :: DayOfWeek -> DayOfWeek
+nextDay (Sunday) = Monday
+nextDay d = succ d
+
+daysToParty :: DayOfWeek -> Int
+daysToParty (Friday) = 0
+daysToParty d 
+    | (fromEnum d) < (fromEnum Friday) = (fromEnum Friday) - (fromEnum d)
+    | otherwise = (abs ((fromEnum Friday) - (fromEnum d))) + (fromEnum Friday)
```

Comment:
You can skip all of the parenthesis here.
Also, you can see that you reuse the same function call many times, so it would make sense to put it into the variables.
Moreover, instead of the manual check, you could use `div .. 7` to get the day 🙂